canonical/loki-k8s-operator
# Added metrics endpoint relation

This commit uses the Prometheus scrape library to implement a
metrics endpoint relation for Loki so that its own metrics can
be scraped by Prometheus.

## Issue
Support self monitoring of Loki.


## Solution
Added a metrics endpoint relation between Loki and Prometheus.


## Context
This commit is part of continue efforts to make the COS stack itself observable.


## Testing Instructions
1. Build and deploy this Loki charm
2. Deploy Prometheus charm from edge
3. Add relation between Prometheus and Loki
4. Check that Loki metrics are queryable from Prometheus dashboard.


## Release Notes
Loki charm now provides a metrics endpoint that can be scraped by Prometheus charm.
